gio/src/subclass/action_map.rs

```diff
@@ -0,0 +1,79 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{Action, ActionMap};
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, GString, IsA, ObjectExt, Quark};
+use once_cell::sync::Lazy;
+
+pub trait ActionMapImpl: ObjectImpl {
+    fn lookup_action(&self, action_map: &Self::Type, action_name: &str) -> Option<Action>;
+    fn add_action(&self, action_map: &Self::Type, action: &Action);
+    fn remove_action(&self, action_map: &Self::Type, action_name: &str);
+}
+
+unsafe impl<T: ActionMapImpl> IsImplementable<T> for ActionMap
+where
+    <T as ObjectSubclass>::Type: IsA<glib::Object>,
+{
+    unsafe extern "C" fn interface_init(
+        iface: glib::ffi::gpointer,
+        _iface_data: glib::ffi::gpointer,
+    ) {
+        let action_map_iface = &mut *(iface as *mut ffi::GActionMapInterface);
+
+        action_map_iface.lookup_action = Some(action_map_lookup_action::<T>);
+        action_map_iface.add_action = Some(action_map_add_action::<T>);
+        action_map_iface.remove_action = Some(action_map_remove_action::<T>);
+    }
+}
+
+static ACTION_MAP_LOOKUP_ACTION_QUARK: Lazy<Quark> =
```

Comment:
I'm wondering if the quark here has to be per action ?

Reply:
Well, no. But you should store a hash table of action_name->results here I guess.